contrib/firebaseio/src/main/java/contrib/LoggingDoFn.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not  use this file except  in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package contrib;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * TODO: Insert description here. (generated by elibixby)
+ */
+public class LoggingDoFn<T> extends DoFn<T, T> {
+
+  private final String className;
+  private transient Logger logger;
+
+  public LoggingDoFn(Class<?> clazz){
+    className = clazz.getName();
+  }
+
+  @Override
+  public void startBundle(Context c) throws Exception {
+    logger = LoggerFactory.getLogger(Class.forName(className));
+  }
+
+  /* (non-Javadoc)
+   * @see DoFn#processElement(com.google.cloud.dataflow.sdk.transforms.DoFn.ProcessContext)
+   */
+  @Override
+  public void processElement(DoFn<T, T>.ProcessContext c) throws Exception {
+    logger.info(c.element().toString());
+    c.output(c.element());
+  }
+}
```

Comment:
Few other weird notes:

- Parameter `level` to let the user specify the log level.
- Use of the `"{}"` instead of `element.toString()` so that we don't convert to string if the log is disabled